demos/coreMQTT_Agent/mqtt_agent_task.c

```diff
@@ -0,0 +1,1025 @@
+            xNetworkResult = prvSocketDisconnect( &xNetworkContext );
+            break;
+        }
+        /* Error. */
+        else
+        {
+            /* Reconnect TCP. */
+            xNetworkResult = prvSocketDisconnect( &xNetworkContext );
+
+            if( xNetworkResult == pdPASS )
+            {
+                xNetworkResult = prvSocketConnect( &xNetworkContext );
+
+                if( xNetworkResult == pdPASS )
+                {
+                    pMqttContext->connectStatus = MQTTNotConnected;
+                    /* MQTT Connect with a persistent session. */
+                    xConnectStatus = prvMQTTConnect( false );
+                }
+            }
+        }
+    } while( xMQTTStatus != MQTTSuccess );
+
+    /* Delete the task if it is complete. */
+    LogInfo( ( "MQTT Agent task completed." ) );
+    vTaskDelete( NULL );
+}
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvConnectToMQTTBroker( void )
+{
+    BaseType_t xNetworkStatus = pdFAIL;
+    MQTTStatus_t xMQTTStatus = MQTTBadParameter;
+
+    /* Connect a TCP socket to the broker. */
+    xNetworkStatus = prvSocketConnect( &xNetworkContext );
+
+    if( xNetworkStatus == pdPASS )
+    {
+        /* Initialize the MQTT context with the buffer and transport interface. */
+        xMQTTStatus = prvMQTTInit();
+
+        if( xMQTTStatus == MQTTSuccess )
+        {
+            /* Form an MQTT connection without a persistent session. */
+            xMQTTStatus = prvMQTTConnect( true );
+        }
+    }
+
+    return ( xMQTTStatus == MQTTSuccess ) ? pdPASS : pdFAIL;
+}
+/*-----------------------------------------------------------*/
+
+static int prvConnectAndCreateDemoTasks( void * pvParameters )
+{
+    MQTTAgentCommandInfo_t xCommandParams = { 0 };
+    uint32_t i, numSuccess = 0;
+
+    ( void ) pvParameters;
+
+    /* Miscellaneous initialization. */
+    ulGlobalEntryTimeMs = prvGetTimeMs();
+
+    /* Set the pParams member of the network context with desired transport. */
+    xNetworkContext.pParams = &secureSocketsTransportParams;
+
+    /* Create the TCP connection to the broker, then the MQTT connection to the
+     * same. */
+    prvConnectToMQTTBroker();
+
+    /* Selectively create demo tasks as per the compile time constant settings. */
+
+    #if ( democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE > 0 )
+        {
+            vStartSimpleSubscribePublishTask( democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE,
+                                              democonfigSIMPLE_SUB_PUB_TASK_STACK_SIZE,
+                                              tskIDLE_PRIORITY,
+                                              taskParameters );
+        }
+    #endif
+
+
+    /* Create an instance of the MQTT agent task. */
+    xTaskCreate( prvMQTTAgentTask,
+                 "MQTT Agent",
+                 configMINIMAL_STACK_SIZE,
+                 NULL,
+                 tskIDLE_PRIORITY + 1,
```

Comment:
Can you add a comment about why the agent task has higher priority than the application tasks?